### ISSUE TITLE
CRUD component does not set height, when Grid#setAllRowsVisible() is used

### ISSUE THREAD

**Issue by timomeinen** [vaadin/flow-components#6654]
### Description of the bug

Using the CRUD component, I want to `setAllRowsVisible(true)`:

    crud.getGrid().setAllRowsVisible(true);

This sets `height: auto` on the Grid via `all-rows-visible`. But the grid is then "cutted" at the bottom by the surrounding `vaadin-crud` element that has a fixed size of `height: 400px`.

### Expected behavior

When enabling `setAllRowsVisible(true)` the CRUD element should also get `height: auto`.

Alternatively, write a delegate on CRUD for `setAllRowsVisible(true)` that sets height on CRUD and calls grid's `setAllRowsVisible(true)`.

### Minimal reproducible example

Create CRUD with 20 elements of data and call:

    crud.getGrid().setAllRowsVisible(true);

### Versions

- Vaadin / Flow version: 24.4.12
- Java version: 21

**Comment by rolfsmeds** [comment]
I don't think it makes sense for CRUD to change its own height based on a setting on the Grid, but I suppose we could add API in CRUD for setting all-rows-visible mode that effectively bases its height on the contents of the Grid. Although I feel that such a mode would not make sense with an editor in the bottom or aside position (i.e. I feel that it only really makes sense with an overlay editor).

**Comment by timomeinen** [comment]
Thank you @rolfsmeds. Makes sense.

I solved it now like this:

    crud.setHeight("auto");
    crud.getGrid().setMinHeight(400, Unit.PIXELS);
    crud.getGrid().setAllRowsVisible(true);
